dcm4che-tool/dcm4che-tool-findscu/src/main/java/org/dcm4che3/tool/findscu/FindSCU.java

```diff
@@ -437,12 +443,30 @@ private static void configureCancel(FindSCU main, CommandLine cl) {
     }
 
     private static void configureKeys(FindSCU main, CommandLine cl) {
-        CLIUtils.addEmptyAttributes(main.keys, cl.getOptionValues("r"));
-        CLIUtils.addAttributes(main.keys, cl.getOptionValues("m"));
+        addFilesAsArgs(main, cl);
+        CLIUtils.addEmptyAttributes(main.keys, main.returnKeys.toArray(new String[main.returnKeys.size()]));
+        CLIUtils.addAttributes(main.keys, main.matchingKeys.toArray(new String[main.matchingKeys.size()]));
         if (cl.hasOption("L"))
             main.addLevel(cl.getOptionValue("L"));
         if (cl.hasOption("i"))
-            main.setInputFilter(CLIUtils.toTags(cl.getOptionValues("i")));
+            main.setInputFilter(CLIUtils.toTags(main.iKeys.toArray(new String[main.iKeys.size()])));
+    }
+
+    private static void addFilesAsArgs(FindSCU main, CommandLine cl) {
+        addFilesAsArgs(main.argList, cl.getOptionValues("r"), main.returnKeys);
+        addFilesAsArgs(main.argList, cl.getOptionValues("m"), main.matchingKeys);
+        addFilesAsArgs(main.argList, cl.getOptionValues("i"), main.iKeys);
+    }
+
+    private static void addFilesAsArgs(List<String> argList, String[] optVals, List<String> keys) {
+        if (optVals != null)
+            for (String key : optVals) {
+                File f = new File(key);
+                if (f.exists())
```

Comment:
Hi Vrinda,
this logic seems too complex. Why do you have to check if the argument is a file?
If option `-r` `-m` or `-i` is used, then then arguments following those options won't be a file, so e.g.:
```
findscu -i PatientName -i ModalitiesInStudy dicomfile.dcm
```
PatientName and ModalitiesInStudy are obviously not files, because they follow `-i`. On the other hand `dicomfile.dcm` is a file.